build.gradle

```diff
@@ -101,6 +103,10 @@ test {
     exclude '**/*Tests.class'
 }
 
+tasks.withType(JavaCompile) {
+    options.compilerArgs += [ '-Xep:DeadException:WARN', '-Xep:GuardedByValidator:OFF' ]
+}
```

Comment:
There's an existing `tasks.withType(JavaCompile)` block on line 59.  Would you mind moving the content of this block to that location to keep everything in one place?

Any reason for overriding the default configuration for the `DeadException` and `GuardedByValidator` rules?  It does not appear we currently have any such violations.

We might want to consider temporarily turning off the `TopLevel` rule (`'-Xep:TopLevel:OFF'`), as that is already covered by Checkstyle.  That will make it easier to focus on the unique issues Error Prone is identifying (20 of 29 issues are `TopLevel` violations).

Reply:
> There's an existing tasks.withType(JavaCompile) block on line 59. Would you mind moving the content of this block to that location to keep everything in one place?

Certainly, that seems like a good cleanup. I'll also add a comment on why we have the new compiler args as well while making that update.

> Any reason for overriding the default configuration for the DeadException and GuardedByValidator rules? It does not appear we currently have any such violations.

Copy/paste from the example docs, I'll remove them if we do not see the need for the exceptions.

> We might want to consider temporarily turning off the TopLevel rule ('-Xep:TopLevel:OFF'), as that is already covered by Checkstyle. That will make it easier to focus on the unique issues Error Prone is identifying (20 of 29 issues are TopLevel violations).

I'm not sure there is that much use in doing so temporarily. The overlap does not really both me. Do you feel strongly on this one at all @ssoloff ? I would suggest we fix the violation directly instead of adding a temporary exception.

Reply:
> Do you feel strongly on this one at all @ssoloff ?

I went back and looked at the existing top-level class violations, and they are all easily fixable (for some reason I thought fixing them would break compatibility).  So don't worry about disabling the `TopLevel` rule; I'll fix those violations soon.